tests/python/pants_test/engine/test_fs.py

```diff
@@ -598,6 +599,44 @@ def test_caches_downloads(self):
           82
         ))
 
+  def generate_original_digest(self) -> Digest:
+    content = b'dummy content'
+    input_files_content = InputFilesContent((
+      FileContent(path='a.txt', content=content),
+      FileContent(path='b.txt', content=content),
+      FileContent(path='c.txt', content=content),
+      FileContent(path='subdir/a.txt', content=content),
+      FileContent(path='subdir/b.txt', content=content),
+      FileContent(path='subdir2/a.txt', content=content),
+      FileContent(path='subdir2/nested_subdir/x.txt', content=content),
+    ))
+    digest = self.request_single_product(Digest, input_files_content)
+    return digest
```

Comment:
Nit: consider directly returning `self.request_single_product(Digest, input_files_content)`. This is subjective, but I don't think the extra line adds much clarity.